x/oracle/keeper/historic_price.go

```diff
@@ -27,6 +27,17 @@ func (k Keeper) HistoricMedians(
 	return medians
 }
 
+// AvailableMedians returns the number of medians in the store for a given denom
+// when called.
+func (k Keeper) AvailableMedians(
+	ctx sdk.Context,
+	denom string,
+) uint32 {
+	medians := k.HistoricMedians(ctx, denom, k.MaximumMedianStamps(ctx))
```

Comment:
avoid data deserialization if not needed. Use native store traversal (not an Iterator function defined in this module) to count elements, rather than doing unnecessary operations

Reply:
If it's an issue, actually we can just remove this function (though I originally requested it) since it's been more efficiently replaced with the extra return on the others.

As far as leverage is using this, I can now avoid calling `AvailableMedians` directly since I'm using `MedianOfHistoricMedians`. (Assuming my other comment is addressed first)

Reply:
Removed `AvailableMedians` method👍